source/extensions/resource_monitors/common/factory_base.h

```diff
@@ -37,33 +37,6 @@ class FactoryBase : public Server::Configuration::ResourceMonitorFactory {
   const std::string name_;
 };
 
-/**
- * Factory for resource monitors that have empty configuration blocks.
- */
-class EmptyConfigFactoryBase : public Server::Configuration::ResourceMonitorFactory {
```

Comment:
Do we have any way of knowing structurally (or enforcing even) that `Empty` is never used? Could we add some reflection based assertion in the factory base to ensure that we always have an empty proto whose `full_name` starts with `envoy.`?

Reply:
Right now the technical problem is that factories are created before protobuf initializes, so we need to change how we provide message names (or face the fiasco).

Reply:
Added a dynamic release assertion banning Empty. We can't check for envoy prefix. since custom filters may live anywhere.